fluent/protocol/chunk.go

```diff
@@ -0,0 +1,116 @@
+package protocol
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/tinylib/msgp/msgp"
+)
+
+func makeChunkID() (string, error) {
+	b, err := uuid.New().MarshalBinary()
```

Comment:
I investigated this question. Performance impact should be low because I use `uuid.EnableRandPool()` here:
 https://github.com/IBM/fluent-forward-go/pull/32/files#diff-4d785a7a013bf7f851c511237c90904dd607ba2cf9f7cbc0a3f100bf08e0df14R35

Rather than creating a new random number every time, `New()` will read a new value from a stream of pooled numbers. You can read more about it here: https://github.com/google/uuid/blob/master/uuid.go#L268-L281

Reply:
I've got a bunch of benchmarks that I'll PR soon, but here is a sample of sending a single message where `Chunk` is called each time.

```
Benchmark_Fluent_Forward_Go_SingleMessageAck-8   	     961	   1139916 ns/op	      90 B/op	       4 allocs/op
```

You can see memory usage is very small.

Reply:
I totally missed `uuid.EnableRandPool()` this, as its already added, we are good.